### ISSUE TITLE
Question for a PhD student

### ISSUE THREAD

**Issue by bongbonglemon** [matsuren/crownconv360depth#1]
I sincerely apologise for abusing 'Issues' to ask a personal question. Its just that I really want to know more about doing research and I think this is the easiest way to get a response from a PhD student him/herself. I'm 20 and will only enter University next year but have been learning deep learning/computer vision on my own (nothing great). My question is:

I have been going through code from several papers on GitHub and they only have .py scripts. At the same time, (because I'm using a MacBook Air) there seems to be only Jupyter Notebook-based cloud solutions (eg. AWS SageMaker, PaperSpace, Google Colab, Kaggle Kernels etc.). I want to learn to code like what researchers are doing now, specifically with regards to how you and other reseachers structure the project: It is typical to see a folder for model and utils respectively. This is something that is made tedious with Jupyter Notebooks as you can't simply import code from another python script. I would like to ask for details on your workflow when it comes to coding in your research project. I would even appreciate if you could just refer me to other resources where you perhaps may have learnt about this yourself.

Once again, I am sorry for abusing 'Issues'. If there is another way to communicate with researchers on GitHub, please do tell.

Yours Sincerely,
A person aspiring to be a deep learning researcher
Gordon

**Comment by matsuren** [comment]
Thank you for your question!

It's great that you start studying deep learning and computer vision before entering University. I can only tell you about my personal experiences here. I hope it will help you.

About my workflow:
Basically, I use jupyter notebook to start writing code (dataset code, training code, network structure, etc.) because it's easier for me to write from scratch. After finishing coding, I convert them into `.py` script manually to train the network. The reason I use `.py` script for training is that `.py` script with arguments is easier to change parameters like batch size, learning rate, total epochs, etc. 

Why only `.py` in Github repo:
I used to commit some jupyter notebooks in Github repo before as you can see [here](https://github.com/matsuren/octDPSNet/tree/master/misc) but version control for jupyter notebook is a little bit messy. So I prefer not to commit notebooks. I would use [Jupytext](https://github.com/mwouts/jupytext) for uploading notebooks in Github. 

Thanks,
